pkg/dtls/conn.go

```diff
@@ -347,8 +347,9 @@ func (c *Conn) notify(level alertLevel, desc alertDescription) {
 
 func (c *Conn) signalHandshakeComplete() {
 	select {
```

Comment:
I think this is racy since signalHandshakeComplete can be called from different routines. Two routines may decide to take the default case at the same time and both try to close the handshakeCompleted channel.

Reply:
ACK.

This will be fixed by merging your `shutdown` work right? I will fix in that PR